l1-contracts/contracts/bridge/asset-tracker/GWAssetTracker.sol

```diff
@@ -466,19 +528,23 @@ contract GWAssetTracker is AssetTrackerBase, IGWAssetTracker {
         require(zkChain != address(0), ChainIdNotRegistered(_chainId));
 
         uint256 settlementLayer = L2_BRIDGEHUB.settlementLayer(_chainId);
-        require(settlementLayer != block.chainid, NotMigratedChain());
+        uint256 readChainMigrationNumber = _getChainMigrationNumber(_chainId);
 
-        uint256 migrationNumber = _getChainMigrationNumber(_chainId);
-        require(assetMigrationNumber[_chainId][_assetId] < migrationNumber, InvalidAssetId(_assetId));
-        uint256 amount = _getOrSaveChainBalance(_chainId, _assetId, migrationNumber);
+        // If the chain already migrated back to GW, then we need the previous migration number.
+        uint256 chainMigrationNumber = settlementLayer == block.chainid
+            ? readChainMigrationNumber - 1
+            : readChainMigrationNumber;
+        require(assetMigrationNumber[_chainId][_assetId] < chainMigrationNumber, InvalidAssetId(_assetId));
+        uint256 amount = _getOrSaveChainBalance(_chainId, _assetId, chainMigrationNumber, true);
```

Comment:
the implementation of `getOrSave` resets to zero only if the value has not been stored yet, so if:
- it was stored during deposit
- you call it here again

it wont be reset, allowing for draining funds on GW